File: src/main/java/com/googongill/aditory/security/jwt/auth/PrincipalDetailsService.java
package com.googongill.aditory.security.jwt.auth;

import com.googongill.aditory.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class PrincipalDetailsService implements UserDetailsService {

    private final UserRepository userRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        return userRepository.findByUsername(username)
                .map(user -> new PrincipalDetails(user))
                .orElseThrow(() -> new UsernameNotFoundException("사용자 " + username + " 를 찾을 수 없습니다."));
    }
}

File: src/main/java/com/googongill/aditory/security/jwt/user/PrincipalDetailsService.kt
package com.googongill.aditory.security.jwt.user

import com.googongill.aditory.domain.User
import com.googongill.aditory.repository.UserRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import java.util.*

@Service
class PrincipalDetailsService(val userRepository: UserRepository) : UserDetailsService {

    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        return Optional.ofNullable(userRepository.findByUsername(username))
            .map { user : User -> PrincipalDetails(user) }
            .orElseThrow { UsernameNotFoundException("사용자 $username 를 찾을 수 없습니다.") }
    }

}
